tests/test_plugin.py

```diff
@@ -2512,9 +2512,7 @@ def test_plugin_shutdown(node_factory):
 
     # Now, should also shutdown or timeout on finish, RPC calls then fail with error code -5
     l1.rpc.plugin_start(p, dont_shutdown=True)
-    needle = l1.daemon.logsearch_start
     l1.rpc.stop()
-    l1.daemon.wait_for_log(r"test_libplugin: shutdown called")
-    l1.daemon.logsearch_start = needle          # we don't know what comes first
-    l1.daemon.is_in_log(r'misc_notifications.py: via lightningd shutdown, datastore failed')
-    l1.daemon.is_in_log(r'test_libplugin: failed to self-terminate in time, killing.')
+    l1.daemon.wait_for_log(r'test_libplugin: shutdown called')
+    l1.daemon.wait_for_logs(['misc_notifications.py: via lightningd shutdown, datastore failed',
+                             'test_libplugin: failed to self-terminate in time, killing.'])
```

Comment:
This broke CI, because wait_for_log gets upset if lightningd dies.  We should note that is why this is done this way, and probably use:

```python
# wait_for_log will get upset if it sees the daemon exit, so do this search manually:
for m in ('test_libplugin: shutdown called', 'misc_notifications.py: via lightningd shutdown, datastore failed', '
    wait_for(lambda: l1.daemon.is_in_log(m))
```

Reply:
Actually, it will work if you just put all three in wait_for_logs.  (That's what the previous code did, with moving the needle back).

Reply:
Thanks, I keep tripping over that same stone !